src/eddymotion/utils.py

```diff
@@ -0,0 +1,150 @@
+
+    Available sorting strategies:
+    - LINEAR: Sorts the items in a linear order.
+    - RANDOM: Sorts the items in a random order.
+    - BVALUE: Sorts the items based on their b-value.
+    - CENTRALSYM: Sorts the items based on their central symmetry.
+    """
+
+    LINEAR = "linear"
+    RANDOM = "random"
+    BVALUE = "bvalue"
+    CENTRALSYM = "centralsym"
+
+
+
+def linear_action(size, **kwargs):
+    """
+    Sort the DWI data volume indices linearly
+
+    Parameters
+    ----------
+    size : :obj:`int`
+        Number of volumes in the dataset
+        (for instance, the number of orientations in a DWI)
+
+    Returns
+    -------
+    :obj:`list` of :obj:`int`
+        The sorted index order.
+
+    Examples
+    --------
+    >>> linear_action(10)
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+    """
+    index_order = np.arange(len(dwdata))
+
+    return index_order
+
+
+def random_action(dwdata, seed=None):
+    """Sort the DWI data volume indices.
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI` DWI dataset, represented by this tool's internal type.
+    seed : :obj:`int` or :obj:`bool`, optional
+        Seed the random number generator. If an integer, the value is used to
+        initialize the generator; if ``True``, the arbitrary value
+        of ``20210324`` is used to initialize it.
+
+    Returns
+    -------
+    index_order : :obj:`numpy.ndarray`
+        The sorted index order.
+    """
+
+    _seed = kwargs.get('seed', None)
+    if seed or seed == 0:
+        _seed = 20210324 if seed is True else seed
+
+    rng = np.random.default_rng(_seed)
+
+    index_order = np.arange(len(dwdata))
+    rng.shuffle(index_order)
+
+    return index_order.to_list()
+
+
+def bvalue_action(dwdata):
+    """
+    Sort the DWI data volume indices in ascending order based on the last column of gradients.
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI` DWI dataset, represented by this tool's internal type.
+
+    Returns
+    -------
+    numpy.ndarray: The sorted index order.
+    """
+    last_column = dwdata.gradients[:, -1]
+    index_order = np.argsort(last_column)
+    return index_order
+
+
+def centralsym_action(size, **kwargs):
+    """
+    Sort the DWI data volume indices in a central symmetric manner.
+
+    Parameters
+    ----------
+    size : :obj:`int`
+        Number of volumes in the dataset
+        (for instance, the number of orientations in a DWI)
+
+    Examples
+    --------
+    >>> centralsym_action(10)
+    [4, 5, 3, 6, 2, 7, 1, 8, 0, 9]
+    >>> centralsym_action(11)
+    [4, 5, 3, 6, 2, 7, 1, 8, 0, 9, 10]
+
+    Returns
+    -------
+    :obj:`list` of :obj:`int`
+        The sorted index order.
+    """
+    linear = list(range(size))
+    half1, half2 = list(reversed(linear[:size // 2])), linear[size // 2:]
+    index_order = [
+        sub[item] for item in range(len(half1))
+        for sub in [half1, half2]
```

Comment:
Okay, I think I caught the problem you were hinting at with odd numbers:

```suggestion
        for sub in [half2, half1]
```

With odd numbers, `half2` is one element longer than `half1`, so starting with an element in `half2` makes more sense. This will require updating the two tests above (added within this review).